src/scripts/main.js

```diff
@@ -1,3 +1,37 @@
 'use strict';
 
-// write your code here
+document.addEventListener('DOMContentLoaded', () => {
+  const logo = document.querySelector('.logo');
+
+  const promise1 = new Promise((resolve) => {
+    logo.addEventListener('click', () => {
+      resolve();
+    });
+  });
+
+  const promise2 = new Promise((resolve, reject) => {
+    setTimeout(() => {
+      reject(new Error('Promise was rejected!'));
+    }, 3000);
+  });
+
+  const onSuccess = () => {
+    const successMessage = document.createElement('div');
+
+    successMessage.className = 'message';
+    successMessage.textContent = 'Promise was resolved!';
+    document.body.appendChild(successMessage);
+  };
+
+  const onError = (error) => {
+    const errorMessage = document.createElement('div');
+
+    errorMessage.className = 'message error-message';
+    errorMessage.textContent = error.message;
+    document.body.appendChild(errorMessage);
+  };
+
+  promise1.then(onSuccess);
+
+  promise2.catch(onError);
```

Comment:
Consider using `Promise.race([promise1, promise2])` to handle the promises together. This way, you can handle whichever promise settles first, whether it's a resolution from clicking the logo or a rejection from the timeout. This approach would prevent both messages from potentially being displayed if the logo is clicked after the timeout.